### ISSUE TITLE
salt-ssh -L with hosts down or unreachable returns wrong results

### ISSUE THREAD

**Issue by ecgg** [saltstack/salt#43449]
### Description of Issue/Question

When a host is down or doesn't resolve salt-ssh -L returns results for the last host which worked.

### Setup
```
# etc/roster
goodboy:
    host:   goodboy.example.com

downboy:
    host: downboy.example.com

noresolve:
    host:   doesntresolveatall.example.com
```

### Steps to Reproduce Issue
```
salt-ssh -L goodboy,downboy,noresolve -r 'uname'

noresolve:
    ----------
    retcode:
        0
    stderr:
        Connection to goodboy closed.
    stdout:
        Linux goodboy 2.6.32-696.6.3.el6.x86_64 #1 SMP Fri Jun 30 13:24:18 EDT 2017 x86_64 x86_64 x86_64 GNU/Linux
downboy:
    ----------
    retcode:
        0
    stderr:
        Connection to goodboy closed.
    stdout:
        Linux goodboy 2.6.32-696.6.3.el6.x86_64 #1 SMP Fri Jun 30 13:24:18 EDT 2017 x86_64 x86_64 x86_64 GNU/Linux
goodboy:
    ----------
    retcode:
        0
    stderr:
        Connection to goodboy closed.
    stdout:
        Linux goodboy 2.6.32-696.6.3.el6.x86_64 #1 SMP Fri Jun 30 13:24:18 EDT 2017 x86_64 x86_64 x86_64 GNU/Linux
```

### Versions Report
Salt Version:
Salt: 2017.7.0

Dependency Versions:
cffi: 1.10.0
cherrypy: Not Installed
dateutil: 2.6.1
docker-py: Not Installed
gitdb: Not Installed
gitpython: Not Installed
ioflo: Not Installed
Jinja2: 2.9.6
libgit2: Not Installed
libnacl: Not Installed
M2Crypto: Not Installed
Mako: Not Installed
msgpack-pure: Not Installed
msgpack-python: 0.4.8
mysql-python: Not Installed
pycparser: 2.18
pycrypto: 2.6.1
pycryptodome: Not Installed
pygit2: Not Installed
Python: 2.7.13 (default, Jun 26 2017, 10:20:05)
python-gnupg: Not Installed
PyYAML: 3.12
PyZMQ: 16.0.2
RAET: Not Installed
smmap: Not Installed
timelib: Not Installed
Tornado: 4.5.1
ZMQ: 4.1.6

System Versions:
dist: fedora 26 Twenty Six
locale: UTF-8
machine: x86_64
release: 4.12.9-300.fc26.x86_64
system: Linux
version: Fedora 26 Twenty Six

**Comment by garethgreenaway** [comment]
Thanks for the report. We'll get someone to look into this asap.

**Comment by rallytime** [comment]
@ecgg This should be fixed with #43889. Can you give that a try?
